### PR TITLE
Add tzdata port and remove prepare-sysroot

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR removes the prepare-sysroot script by adding a port of `tzdata` and moving the remaining tasks to the `base-configuration` package. The reason I had to host the tarball for `tzdata` myself is that iana doesn't pack their tarballs properly, so all the files were spilled.

Blocked on a `xbstrap` release with managarm/xbstrap#25

### CODE REVIEW

File: bootstrap.d/sys-libs.yml [hunk 1]
```diff
@@ -103,3 +103,62 @@ packages:
...
+      - args: ['/usr/sbin/zic', '-L', '@THIS_SOURCE_DIR@/leapseconds', '-d', '@THIS_COLLECT_DIR@/usr/share/zoneinfo/right', '@THIS_BUILD_DIR@/pacificnew']
+      - args: ['/usr/sbin/zic', '-L', '/dev/null', '-d', '@THIS_COLLECT_DIR@/usr/share/zoneinfo', '@THIS_BUILD_DIR@/systemv']
+      - args: ['/usr/sbin/zic', '-L', '/dev/null', '-d', '@THIS_COLLECT_DIR@/usr/share/zoneinfo/posix', '@THIS_BUILD_DIR@/systemv']
+      - args: ['/usr/sbin/zic', '-L', '@THIS_SOURCE_DIR@/leapseconds', '-d', '@THIS_COLLECT_DIR@/usr/share/zoneinfo/right', '@THIS_BUILD_DIR@/systemv']
+      # Copy some needed files to their location
+      - args: ['cp', '@THIS_BUILD_DIR@/zone.tab', '@THIS_BUILD_DIR@/zone1970.tab', '@THIS_BUILD_DIR@/iso3166.tab', '@THIS_COLLECT_DIR@/usr/share/zoneinfo']
+      # Create the posixrules file, POSIX requires daylight saving rules to be in accordance with US rules, thus use New York
+      - args: ['/usr/sbin/zic', '-d', '@THIS_COLLECT_DIR@/usr/share/zoneinfo', '-p', 'America/New_York']
+      # Copy /etc/localtime from the host
+      - args: ['rsync', '-a', '/etc/localtime', '@THIS_COLLECT_DIR@/etc/localtime']
```
Comment (ArsenArsen) [inline bootstrap.d/sys-libs.yml:164]: should not be in any package, unless maybe by being set to UTC (to not use build server settings or whatever), and set as a [config file to `xbps-create`](https://man.voidlinux.org/xbps-create.1#F,) (requires work in xbstrap)

Comment (Dennisbonke) [reply bootstrap.d/sys-libs.yml:164]: Yeah you're right, I'll just leave a note there saying this should be fixed while defaulting to utc for the time being

**Comment by avdgrinten** [comment]
This PR needs to be updated to extract the upstream tarball directly (and that is blocked on a new xbstrap release).

I also have some other concerns: what if `zic` is not installed on the host system? `zic` needs to be listed as a dependency, right? Is it commonly installed on Linux systems?

**Comment by Dennisbonke** [comment]
Well, assuming you have a libc, then yes, `zic` should be installed. On debian, it's part of `libc-bin`
```
dennis@DENNIS-MAIN-LAPTOP-SIDUCTION:~$ dpkg -S /usr/sbin/zic 
libc-bin: /usr/sbin/zic
dennis@DENNIS-MAIN-LAPTOP-SIDUCTION:~$
```

**Comment by Dennisbonke** [comment]
The last commit addresses the url, and this means that this pr is now blocked on a new `xbstrap` release that has managarm/xbstrap#25

**Comment by avdgrinten** [comment]
Alright, LGTM. Thanks you!

**Pull request merged**